crates/modelardb_manager/src/remote.rs

```diff
@@ -119,13 +217,46 @@ impl FlightService for FlightServiceHandler {
         Err(Status::unimplemented("Not implemented."))
     }
 
-    /// TODO: Provide the schema of a table in the catalog. The name of the table must
-    ///       be provided as the first element in `FlightDescriptor.path`.
+    /// Provide the schema of a table in the catalog. The name of the table must be provided as the
+    /// first element in `FlightDescriptor.path`.
     async fn get_schema(
         &self,
-        _request: Request<FlightDescriptor>,
+        request: Request<FlightDescriptor>,
     ) -> Result<Response<SchemaResult>, Status> {
-        Err(Status::unimplemented("Not implemented."))
+        let flight_descriptor = request.into_inner();
+        let table_name = flight_descriptor
+            .path
+            .get(0)
+            .ok_or_else(|| Status::invalid_argument("No table name in FlightDescriptor.path."))?;
+
+        let table_sql = self
+            .context
+            .metadata_manager
+            .table_sql(table_name)
+            .await
+            .map_err(|error| {
+                Status::not_found(format!(
+                    "Table with name '{table_name}' does not exist: {error}",
+                ))
+            })?;
+
+        // Parse the SQL and perform semantic checks to extract the schema from the statement.
```

Comment:
We cannot store the schema like in the server since the server stores empty parquet files on disk to save the schema. We could save it in the metadata database like we do for model tables already but since we already need to store the SQL to allow for initializing the database schema when servers are registered, I think it makes sense to also use the SQL here.

Getting the schema should be a fairly rare operation and parsing the SQL should not be that expensive so i think it would be overkill to add another column to the `table_metadata` table just for this.